### ISSUE TITLE
Allow product to be based on a template coupon

### ISSUE THREAD

**Issue by Jon007** [MarieComet/wcs-sell-coupons#8]
For maximum flexibility, the product could be linked to a template coupon which could be chosen from the product admin.

This would allow coupons based on special rules to be sold.

Which could be to allow gift of a discount on a particular category, or other rule.

**Comment by MarieComet** [comment]
Hi @Jon007, big thanks for the commits !
I take note of your ideas, they could be good improvements for this plugin.
Unfortunately I do not think I have time to develop them in the months to come.
Feel free to make some PR, I will merge them on the main branch.

Marie.

**Comment by Jon007** [comment]
Sure, thanks for putting this together, I had been looking for something like this and had the same experience as you noted on your blog.

I also don't really have time either.. !

We were considering the idea for customer rewards.
There are coupon extensions which can gift products to a customer based on rules, automatically adding the gift products to the basket.

However the shop wants to gift customer a coupon for their next order to encourage the customer to return.

One implementation for this would be to auto-add a coupon product to their order so that when the order is paid, another coupon is issued, with a time-limit, to encourage repeat custom.

Several of the other ideas logged relate to this.

However we may not implement this for now, until we have the use case clear.. 
it may be better for the business to issue the coupon on delivery rather than on purchase...  or add coupon note to delivery slip to include with package..

**Comment by MarieComet** [comment]
Did you look at the WooThemes plugin "WooCommerce Smart Coupons" ? Maybe some of its features can solve these needs.

**Comment by Jon007** [comment]
Yes I'm aware of WooCommerce Smart Coupons, it's probably quite good, have you tried it?

If there there is a lot of demand for the gift vouchers, maybe.  If we only sell 10 then no.
If you spend $99/year for every tool you might need, that could rapidly become uneconomic and somewhat defeats the purpose of an open source solution.   

Actually we do use https://wordpress.org/plugins/woocommerce-auto-added-coupons/
(PRO 29EUR) and have already integrated it with multi-lingual solution etc, so anything else we want to do is only a few lines of extra code.
